src/browserContext.ts

```diff
@@ -103,6 +103,19 @@ export abstract class BrowserContextBase extends platform.EventEmitter implement
   abstract route(url: types.URLMatch, handler: network.RouteHandler): Promise<void>;
   abstract close(): Promise<void>;
 
+  async grantPermissions(permissions: string[], options?: { origin?: string }) {
+    const origin = options && options.origin || '*';
+    const existing = new Set(this._permissions.get(origin) || []);
+    permissions.forEach(p => existing.add(p));
+    const list = [...existing.values()];
+    this._permissions.set(origin, list);
+    await this._doGrantPermissions(origin, list);
+  }
+
+  async clearPermissions() {
+    this._permissions.clear();
```

Comment:
It is confusing that we don't call `_doClearPermissions` here.

Reply:
I thought the same way. Fixing.